source/agora/node/EnrollmentManager.d

```diff
@@ -0,0 +1,469 @@
+/*******************************************************************************
+
+    Contains supporting code for enrollment process.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.EnrollmentManager;
+
+import agora.common.Amount;
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr : Pair;
+import agora.common.crypto.Schnorr : Signature;
+import agora.common.crypto.Schnorr : sign;
+import agora.common.Deserializer;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.common.Types;
+import agora.consensus.data.EnrollmentData;
+
+import d2sqlite3.database;
+import d2sqlite3.library;
+import d2sqlite3.results;
+import d2sqlite3.sqlite3;
+
+import ocean.util.log.Logger;
+
+import std.algorithm;
+import std.conv : to;
+import std.file;
+import std.path;
+import std.random;
+import std.stdio;
+
+/*******************************************************************************
+
+    Initialize the enrollment manager
+
+    Enrollment manager initialization should only be done once per process,
+    but `loggerCallback` uses `log` which is a TLS variable that might not
+    be initialized if 'loggerCallback' was to be called immediately.
+ 
+*******************************************************************************/
+
+static this ()
+{
+    log = Log.lookup(__MODULE__);
+    EnrollmentManager.initialize();
+}
+
+/// Logger instance
+private Logger log;
+
+/*******************************************************************************
+    Handles data and process for enrollment
+*******************************************************************************/
+
+public class EnrollmentManager
+{
+    /***************************************************************************
+
+        Initialization function
+
+        Must be called once per process (not per thread!),
+        before the class can be used.
+
+    ***************************************************************************/
+
+    public static void initialize ()
+    {
+        static import core.atomic;
+        static shared bool initialized;
+        if (core.atomic.cas(&initialized, false, true))
```

Comment:
The SQLite library is already initialized in another place. It shouldn't be initialized twice. I suggest removing this code.